### PR TITLE
improve parse_mtree

### PR DESCRIPTION
**Pull request by Wenzel**
This commit improves the parsing of `mtree_output` by parsing each line and relying on `strtok_r`.

If think it's a cleaner implementation.

### CODE REVIEW

**Comment by namidairo** [comment]
I was thinking of just fixing up the old one to something along these lines (Do I even bother changing strtoll to strtoull? I don't think anyone is going to have 7 exabytes? addressable on a single system for a while)

```static addr_t
parse_mtree(char *mtree_output)
{
    char *ptr;

    if(NULL == (ptr = strstr(mtree_output, "alias ram-above-4g")))
    return 0;

    int i = 0;
    
    for(i = 0; i <= 40; i++) {
      if(ptr[0] == '-') {
        ptr++;
        return (addr_t) strtoll(ptr, (char **) NULL, 16) + 1;
      }
      ptr--;
    }
    
    return 0;
}

**Comment by Wenzel** [comment]
Don't bother with 7 exabytes, the patch is just temporary anyway.
What about my implementation ?
i was trying to remote the hardcoded `50` (or `40`) here and have a more logical parsing
- for each line try to find `above-4g`
- find the number from the beginning of the line
- convert it and return

**Comment by Wenzel** [comment]
Up,
what do you think ?
Can we merge this ?

File: libvmi/driver/kvm/kvm.c [hunk 1]
```diff
@@ -288,26 +288,52 @@ parse_seg_reg_value(
...
+            break;
+        }
+        // consume r\n
+        line = strtok_r(NULL, "n", &tmp);
+        if (line == NULL)
+            return 0;
+    } while ((line = strtok_r(NULL, line_delim, &tmp)) != NULL);
 
-    goto error;
+    // did we found above 4g ?
```
Comment (namidairo) [inline libvmi/driver/kvm/kvm.c]: Did you mean find?

**Comment by namidairo** [comment]
Rest of it looks fine, if only to get rid of the dependence on magic numbers. I'll merge it after the minor comment thing and then we'll see what Travis CI says.

File: libvmi/driver/kvm/kvm.c (hunk 1 above)
Comment (Wenzel) [reply libvmi/driver/kvm/kvm.c]: yes, sorry, i'll fix it right away !

**Pull request merged**